ecs-cli/modules/clients/aws/ecs/client.go

```diff
@@ -287,11 +287,20 @@ func cachedTaskDefinitionRevisionIsActive(cachedTaskDefinition *ecs.TaskDefiniti
 // constructTaskDefinitionCacheHash computes md5sum of the region, awsAccountId and the requested task definition data
 // BUG(juanrhenals) The requested Task Definition data (taskDefinitionRequest) is not created in a deterministic fashion because there are maps within
 // the request ecs.RegisterTaskDefinitionInput structure, and map iteration in Go is not deterministic. We need to fix this.
+// FIXED(pzimmermann) using a marshalled json representation for the task definition data to be deterministic
 func (c *ecsClient) constructTaskDefinitionCacheHash(taskDefinition *ecs.TaskDefinition, request *ecs.RegisterTaskDefinitionInput) string {
 	// Get the region from the ecsClient configuration
 	region := aws.StringValue(c.params.Session.Config.Region)
-	awsUserAccountID := utils.GetAwsAccountIdFromArn(aws.StringValue(taskDefinition.TaskDefinitionArn))
-	tdHashInput := fmt.Sprintf("%s-%s-%s", region, awsUserAccountID, request.GoString())
+	awsUserAccountId := utils.GetAwsAccountIdFromArn(aws.StringValue(taskDefinition.TaskDefinitionArn))
+	sortedRequestString, err := utils.SortedGoString(request)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error": err,
+		}).Warn("Error during json marshalling; fallback to non-deterministic task definition data used for cache hash")
+		tdHashInput := fmt.Sprintf("%s-%s-%s", region, awsUserAccountId, request.GoString())
```

Comment:
This looks a bit redundant, what do you think of changing this to the following, then line 302 and line 303 are not needed
```
if err != nil {
    log.WithFields(....)
    sortedRequestString = request.GoString()
}
```